File: producteur/lib/models/Producteur.dart
class Producteur{
  String? producterfirstname;
  String? productersecondname;
  String? producterphone;
  String? producterpassword;


  Producteur({
    this.producterfirstname,
    this.productersecondname,
    this.producterphone,
    this.producterpassword
  });

  Producteur.fromJson(Map<String , dynamic> json){
    producterfirstname=json['producter_firstname'];
    productersecondname=json['producter_secondname'];
    producterphone=json['producter_phone'];
    producterpassword=json['producter_password'];

  }

  Map<String,dynamic> toJson(){
    final Map<String, dynamic> data = Map<String, dynamic>();
    data['producter_firstname']=producterfirstname;
    data['producter_secondname']=productersecondname;
    data['producter_phone']=producterphone;
    data['producter_password']=producterpassword;

    return data;
  }

}

File: Backend Producteur/Backend 1/models/producteur.js
const db = require('../configuration/database');
const bcrypt = require('bcrypt');

class Producteur {


    static createProducter(prodData, callback) {
        const { producterFirstName, producterSecondName, producterPhone, producterPassword } = prodData;
        
        // Hacher le mot de passe avant de le sauvegarder dans la base de données
        bcrypt.hash(producterPassword, 10, (err, hashedPassword) => {
            if (err) return callback(err);

            const query = `
                INSERT INTO producter (producter_firstname, producter_secondname, producter_phone, producter_password)
                VALUES (?, ?, ?, ?)
            `;
            db.query(query, [producterFirstName, producterSecondName, producterPhone, hashedPassword], (err, result) => {
                if (err) return callback(err);
                callback(null, result);
            });
        });
    }
    

    static findByPhone(prodtData, callback) {

        const { producterPhone ,producterPassword } = prodtData;
        const query = `SELECT * FROM producter WHERE producter_phone = ?`;
        db.query(query, [producterPhone], (err, results) => {
            if(err){
                callback(err);
            }else if(results.length>0){
                const producter = results[0];
                bcrypt.compare(producterPassword, producter.producter_password, (err, result)=>{

                    if(result){
                        callback(null, producter);
                    }else{
                        callback(null, null);
                    }
                });
            }else{
                callback(null, null);
            }
        })
    }


}

module.exports = Producteur;